src/Agent.Worker/Release/ContainerFetchEngine/IContainerProvider.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release.ContainerFetchEngine
+{
+    // NOTE: FetchEngine specific interface shouldn't take dependency on Agent code.
```

Comment:
It's a good constraint to have if we were to move the download functionality into a separate EXE. However let's target this goal subsequently, not needed right now.